picard/ui/mainwindow.py

```diff
@@ -147,6 +147,9 @@ def keyPressEvent(self, event):
     def show(self):
         self.restoreWindowState()
         super().show()
+        if config.setting['check_for_updates'] and datetime.date.today().toordinal() >= config.persist['last_update_check'] + config.setting['update_check_days']:
```

Comment:
`datetime.date.today().toordinal() >= config.persist['last_update_check'] + config.setting['update_check_days']`

If `update_check_days` is 0, and `>=` is used then it will check more than once per day.

So i think it should be `>` + ensure `update_check_days` is >= 0
Or `>=` and ensure `update_check_days` is >= 1

Reply:
With the zero interval I had intended to allow multiple checks per day, but I see now that might not be appropriate or required in a production environment.  I think your second suggested option is better because it allows checks on successive days with the 'update_check_days' setting at 1, whereas the first option will only do checks on successive days if the setting is 0.  It all comes down to how you interpret the meaning of 'update_check_days'.  The interpretation when I was writing it was that a value of 1 meant to check every 1 day (i.e.: every day).  Following this logic, I also suggest that the lower limit in the settings be changed from 0 to 1 for the 'update_check_days' setting.